extension/registry.go

```diff
@@ -24,6 +24,9 @@ import (
 // implementation.
 type Registry interface {
 
+	// GetAdminStorage returns the storage.AdminStorage to be used.
+	GetAdminStorage() storage.AdminStorage
```

Comment:
Actually, we've been planning on getting rid of the error in the other methods. It ties in with a Registry refactor @RJPercival has been preparing.

The TL;DR is that Registry becomes a struct, so all errors go away (either you successfully create one or you don't).